edb/server/config/ops.py

```diff
@@ -68,15 +69,25 @@ class Operation(NamedTuple):
     setting_name: str
     value: Union[str, int, bool, Collection[Union[str, int, bool, None]], None]
 
-    def get_setting(self, spec: spec.Spec):
+    def get_setting(self, spec: spec.Spec) -> Optional[spec.Setting]:
         try:
             return spec[self.setting_name]
         except KeyError:
-            raise errors.ConfigurationError(
-                f'unknown setting {self.setting_name!r}') from None
+            if self.scope == qltypes.ConfigScope.GLOBAL:
+                return None
+            else:
+                raise errors.ConfigurationError(
+                    f'unknown setting {self.setting_name!r}') from None
 
-    def coerce_value(self, setting: spec.Setting, *,
+    def coerce_value(self, setting: Optional[spec.Setting], *,
                      allow_missing: bool = False):
+        # A missing setting indicates that this is a global, not a setting
+        if not setting:
```

Comment:
This smells. Do we have to reuse this machinery for globals?

Reply:
I think overall reusing the Operation machinery here is good, but I rearranged the logic here to be driven by the `scope`, and never attempting to look up a `Setting` for a global, which I think smells a lot less